### PR TITLE
Add a legacy Heartbeat to Kubernetes clusters to maintain support for older clients

### PR DESCRIPTION
**Pull request by tigrato**
This PR introduces an extra heartbeat to maintain compatibility with older clients that do not support the new `KubeServer` feature.

### CODE REVIEW

File: lib/kube/proxy/server.go [hunk 1]
```diff
@@ -118,6 +118,10 @@ type TLSServer struct {
 	// reconcileCh triggers reconciliation of proxied kube_clusters.
 	reconcileCh chan struct{}
 	log         *logrus.Entry
+	// legacyHeartBeat is used to heartbeat clusters as KindKubeService in order to support older
```
Comment (r0mant) [inline lib/kube/proxy/server.go:121]:
```suggestion
	// legacyHeartbeat is used to heartbeat clusters as KindKubeService in order to support older
```

**Review by r0mant — COMMENTED** [review]
I know it will be gone in Teleport 12 but can we still add test coverage for this?

**Review by r0mant — APPROVED** [review]

**Review by espadolini — APPROVED** [review]
This addresses v10 clients using a v11 `kubernetes_service`, but will v10 `kubernetes_service` agents still work as intended?

**Comment by tigrato** [comment]
@espadolini yes. That part was already covered in the original PR.

**Pull request merged**